frontend/coprs_frontend/coprs/logic/complex_logic.py

```diff
@@ -97,34 +98,43 @@ def delete_copr(cls, copr, admin_action=False):
         else:
             user = flask.g.user
 
-        builds_query = BuildsLogic.get_multiple_by_copr(copr=copr)
+        builds = [x.id for x in BuildsLogic.get_multiple_by_copr(copr=copr)]
 
         if copr.persistent:
             raise exceptions.InsufficientRightsException("This project is protected against deletion.")
 
-        for build in builds_query:
+        for chunk in batched(builds, 1000):
             # Don't send delete action for each build, rather send an action to delete
             # a whole project as a part of CoprsLogic.delete_unsafe() method.
-            BuildsLogic.delete_build(user, build, send_delete_action=False)
+            BuildsLogic.delete_builds(user, chunk, send_delete_action=False)
```

Comment:
Where do we commit? At first glance, it seems we don't, and we still create a huge transaction.

Reply:
We commit in the